pkg/controller/bootstrap/bootstrap.go

```diff
@@ -63,28 +67,35 @@ func (b *Bootstrap) Run(destDir string) error {
 			continue
 		}
 
-		path := filepath.Join(b.manifestDir, info.Name())
-		raw, err := ioutil.ReadFile(path)
+		file, err := os.Open(filepath.Join(b.manifestDir, info.Name()))
 		if err != nil {
-			return err
+			return fmt.Errorf("error opening %s: %v", file.Name(), err)
 		}
+		defer file.Close()
 
-		obji, err := runtime.Decode(scheme.Codecs.UniversalDecoder(v1.SchemeGroupVersion), raw)
+		manifests, err := parseManifests(file.Name(), file)
 		if err != nil {
-			glog.V(4).Infof("skipping path because of error: %v", err)
-			// don't care
-			continue
+			return fmt.Errorf("error parsing manifests from %s: %v", file.Name(), err)
 		}
 
-		switch obj := obji.(type) {
-		case *v1.MachineConfigPool:
-			pools = append(pools, obj)
-		case *v1.MachineConfig:
-			configs = append(configs, obj)
-		case *v1.ControllerConfig:
-			cconfig = obj
-		default:
-			glog.Infof("skipping %q %T", path, obji)
+		for idx, m := range manifests {
+			obji, err := runtime.Decode(scheme.Codecs.UniversalDecoder(v1.SchemeGroupVersion), m.Raw)
+			if err != nil {
+				glog.V(4).Infof("skipping path %q [%d] manifest because of error: %v", file.Name(), idx+1, err)
+				// don't care
+				continue
+			}
+
+			switch obj := obji.(type) {
+			case *v1.MachineConfigPool:
+				pools = append(pools, obj)
+			case *v1.MachineConfig:
+				configs = append(configs, obj)
+			case *v1.ControllerConfig:
+				cconfig = obj
+			default:
+				glog.Infof("skipping %q [%d] manifest because %T", file.Name(), idx+1, obji)
```

Comment:
`s/because %T/of unhandled type %T/` ?

Reply:
we are skipping because of the type, not `unhandled`